src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -179,30 +179,36 @@ public CSVFileScanFragment(
   @Override
   protected TupleBatch fetchNextReady() throws IOException, DbException {
     long lineNumberBegin = lineNumber;
+    boolean unfinishedEncapsulatedTokenRow = false;
 
     while ((buffer.numTuples() < TupleBatch.BATCH_SIZE) && !flagAsIncomplete) {
       lineNumber++;
       if (parser.isClosed()) {
         break;
       }
 
+      if (unfinishedEncapsulatedTokenRow) {
+        onLastRow = true;
+      }
+
       try {
         if (!onLastRow) {
           record = iterator.next();
         }
       } catch (Exception e) {
```

Comment:
We definitely should not be swallowing all exceptions here. Why not call `iterator.hasNext()` before `iterator.next()`, so we know the latter won't throw (and if it does, we shouldn't catch it since it likely isn't recoverable)? To be precise, exception handling for the truncated quoted string case would still be done in the `catch` block for  `iterator.hasNext()`, but `iterator.next()` wouldn't be called in a `try` block (since we're not interested in recovering from an unexpected exception).